[build,lint] add eslint workflow

I had to remove the lint.yml workflow due to the following error that __only happens in CI__:

```

> @narmi/design_system@2.11.1 lint
> eslint "./src/**" "./scripts/**" "./tokens/**" --no-error-on-unmatched-pattern

There was a problem loading formatter: /home/runner/work/design_system/design_system/node_modules/eslint/lib/cli-engine/formatters/stylish
Error: Cannot find module 'text-table'
Require stack:
- /home/runner/work/design_system/design_system/node_modules/eslint/lib/cli-engine/formatters/stylish.js
- /home/runner/work/design_system/design_system/node_modules/eslint/lib/cli-engine/cli-engine.js
- /home/runner/work/design_system/design_system/node_modules/eslint/lib/eslint/eslint.js
- /home/runner/work/design_system/design_system/node_modules/eslint/lib/eslint/index.js
- /home/runner/work/design_system/design_system/node_modules/eslint/lib/cli.js
- /home/runner/work/design_system/design_system/node_modules/eslint/bin/eslint.js
Error: Process completed with exit code 2.
```

This ticket is to add an eslint workflow that actually works.

--- a/.github/workflows/lint.yaml
+++ b/.github/workflows/lint.yaml
@@ -1,0 +1,22 @@
+name: Lint
+on: [pull_request]
+jobs:
+  run-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16
+          registry-url: https://registry.npmjs.org
+
+      - name: eslint
+        run: |
+          echo "The job was automatically triggered by a ${{ github.event_name }} event."
+          npm ci
+          npm run lint

